packages/medusa/src/services/__tests__/sales-channel.ts

```diff
@@ -92,4 +92,61 @@ describe("SalesChannelService", () => {
       })
     })
   })
+
+  describe("delete", () => {
+    const salesChannelData = {
+      name: "sales channel 1 name",
+      description: "sales channel 1 description",
+      is_disabled: false,
+    }
+
+    const salesChannelRepositoryMock = MockRepository({
+      findOne: jest.fn().mockImplementation((queryOrId: string | FindOneOptions<SalesChannel>): any => {
+        return Promise.resolve({
+          id:
+            typeof queryOrId === "string"
+              ? queryOrId
+              : ((queryOrId?.where as FindConditions<SalesChannel>)?.id ?? IdMap.getId("sc_adjhlukiaeswhfae")),
+          ...salesChannelData
+        })
+      }),
+      softRemove: jest.fn().mockImplementation((id: string): any => {
+        return Promise.resolve()
+      }),
```

Comment:
should this be added to the repository in the outer describe and that repo mock be used instead?